### PR TITLE
If no geofence is found map then use default value

### PR DESCRIPTION
**Pull request by brianmay**
This is required because the default value is "" which tells the mqtt
server to delete the retained value.

This in turn means that applications that care about when the car is in
a unknown location will not get the retained value.

This is a bit different from all of the other values in
`@always_published`. For them "" is only published if the value is
unknown or an error occurred. So it makes sense to delete the retained
value.

In this case however, a value of "" generally means "the car is outside
any known geofence" which is a valid condition it itself, and it should
be retained.

There was speculation this might affect the shift_state, but
unfortunately this was not followed up with any evidence or details. #1912 

There are any number of solutions that could be deployed here. The
advantage of this solution however is it is not going to break anything
unless you specifically define the TESLAMATE_DEFAULT_GEOFENCE variable.

Fixes: #1336

### CODE REVIEW

**Comment by brianmay** [comment]
Still in the process of testing this to make sure it works. I think I just noticed a problem already, will fix.

This perhaps isn't the perfect solution. But this solution unlike others I can think of avoids breaking the existing API.

**Comment by brianmay** [comment]
As far as I can tell that test failure is unrelated to anything I did...

**Comment by brianmay** [comment]
Tests have been successful so far. Set `TESLAMATE_DEFAULT_GEOFENCE` to "nowhere" and Tesla reported it was "nowhere" instead of "".

File: config/runtime.exs [hunk 1]
```diff
@@ -80,6 +80,9 @@ defmodule Util do
   end
 end
 
+config :teslamate,
+  default_geofence: System.get_env("TESLAMATE_DEFAULT_GEOFENCE")
```
Comment (adriankumpf) [inline config/runtime.exs:84]: Let's drop the `TESLAMATE_` prefix for the sake of consistency.

**Review by adriankumpf — CHANGES_REQUESTED** [review]
That's pragmatic solution to this long-standing issue 👍

Could you add a brief description of that env variable to `website/docs/configuration/environment_variables.md`?

**Comment by adriankumpf** [comment]
> As far as I can tell that test failure is unrelated to anything I did.
> 
> LATER: I notice other PRs are failing on the same test. So probably a flaky test I think.

Yup, there are a bunch of flaky tests that really need to be fixed

**Comment by brianmay** [comment]
> Could you add a brief description of that env variable to `website/docs/configuration/environment_variables.md`?

Done.

**Pull request merged**